### ISSUE TITLE
Issue with transit timing/duration compared to Exoplanet?

### ISSUE THREAD

**Issue by catrionamurray** [exoplanet-dev/exo4jax#3]
Using my slightly hacked code (https://github.com/exoplanet-dev/exo4jax/issues/2) when I get the light curve using the parameter in that issue the transit seems offset in time (and also with a different transit duration) compared to the same params in the original `exoplanet`?:

exoplanet:
![exo](https://user-images.githubusercontent.com/34891147/190796704-99d8b56d-e059-420f-87d5-9df9bd320238.png)

exo4jax:
![exo4jax](https://user-images.githubusercontent.com/34891147/190796748-de1ff3ed-6dc2-450f-a2fa-348eabaaf531.png)

**Comment by catrionamurray** [comment]
This could potentially be related to the difference in x,y,z I get between exoplanet and exo4jax?:
![exo_pos](https://user-images.githubusercontent.com/34891147/190806110-f72a0746-9e37-4901-8e74-f0d8cfe75b67.png)
![exo4jax_pos](https://user-images.githubusercontent.com/34891147/190806123-2dc07683-0722-4282-8c1f-d78e904c46eb.png)

**Comment by dfm** [comment]
Thanks! There could well be a bug! Can you share exactly the code you're using in each case to help track it down?

**Comment by catrionamurray** [comment]
I've potentially further tracked it down to a difference in semi-major axis between the two programs? Exoplanet is showing 8.69, exo4jax shows 4.18. If I set the semi-major axis to 8.69 it also solves this issue!

**Comment by catrionamurray** [comment]
Exo4jax code:

mstar = 0.98
rstar = 0.93
time_transit = 0.0
period = 1
b = 0.1
rr = 0.1
ld = [0.1,0.3]
t = np.linspace(-0.15,0.15,1000)

`host_star = orbits.KeplerianCentral.init(mass=mstar,radius=rstar)`

`planet = orbits.KeplerianBody.init(
        central=host_star,
        time_transit=time_transit,
        period=period,
        impact_param=b,
        radius=jnp.array(rr * rstar))`
    
`orbit = orbits.KeplerianOrbit(planet)`
`lc = light_curves.QuadLightCurve(ld[0], ld[1])`
`# lc_to_return = lc.light_curve(planet, t)`

`r_star = planet.central_radius`
`x, y, z = planet.relative_position(t)`
`b = jnp.sqrt(x**2 + y**2) / r_star`
`r = planet.radius / r_star`
`lc_func = partial(light_curve, lc.u1, lc.u2, order=10)`
`lc_to_return = jax.vmap(lc_func)(b, jnp.array([r] * len(b)))`
`lc_to_return = jnp.where(z > 0, lc_to_return, 0)`

**Comment by catrionamurray** [comment]
Exoplanet code:

mstar = 0.98
rstar = 0.93
time_transit = 0.0
period = 1
b = 0.1
rr = 0.1
ld = [0.1,0.3]
t = np.linspace(-0.15,0.15,1000)

`orbit = xo.orbits.KeplerianOrbit(
                    period=period,
                    t0=time_transit,
                    b=b,
                    r_star=rstar,
                    m_star=mstar
)`

`lc = (
    xo.LimbDarkLightCurve(*ld)
    .get_light_curve(orbit=orbit,r=rr*rstar,t=t)
    .eval()
)`

**Comment by catrionamurray** [comment]
> I think this is potentially related to the difference in x,y,z I get between exoplanet and exo4jax?: ![exo_pos](https://user-images.githubusercontent.com/34891147/190806110-f72a0746-9e37-4901-8e74-f0d8cfe75b67.png) ![exo4jax_pos](https://user-images.githubusercontent.com/34891147/190806123-2dc07683-0722-4282-8c1f-d78e904c46eb.png)
> 
> If I put the x,y,z I get from exoplanet into the calculation of `b` instead of using `.relative_position(t)` then I get the same LC plot! Are they meant to be different and I'm misunderstanding what x/y/z are or is there a bug?

I actually uploaded the wrong plot for exoplanet:
![exo_pos2](https://user-images.githubusercontent.com/34891147/190828196-3885a268-f45c-483c-8272-77e64b4ddc43.png)

**Comment by catrionamurray** [comment]
I've solved the issue! So if I put period in as an integer (period=1 in my case) then `M0 = jnp.full_like(period, 0.5*jnp.pi)` (line 216 of keplerian.py) will make the array of the same data type as the provided array. So it rounds down to 1, which is why there was a difference. If I change period = 1 to period = 1.0 it works as expected. I think if you add the dtype=float or something equivalent, or convert period to a float somewhere, it would avoid this issue, though obviously it's not super physical for an integer period...

**Comment by dfm** [comment]
Wow - great debugging!! I'll look into making sure that doesn't happen next week!

**Comment by dfm** [comment]
Let's keep this open to remember to fix it :D

**Comment by catrionamurray** [comment]
Thanks Dan! I'll test out the newest version once it's merged!
